includes/amp-helper-functions.php

```diff
@@ -464,38 +464,43 @@ function amp_register_default_scripts( $wp_scripts ) {
 		]
 	);
 
-	// Get all AMP components as defined in the spec.
-	$extensions = [];
+	// Register all AMP components as defined in the spec.
 	foreach ( AMP_Allowed_Tags_Generated::get_allowed_tag( 'script' ) as $script_spec ) {
-		if ( isset( $script_spec[ AMP_Rule_Spec::TAG_SPEC ]['extension_spec']['name'], $script_spec[ AMP_Rule_Spec::TAG_SPEC ]['extension_spec']['version'] ) ) {
-			$versions = $script_spec[ AMP_Rule_Spec::TAG_SPEC ]['extension_spec']['version'];
-			array_pop( $versions );
-			$extensions[ $script_spec[ AMP_Rule_Spec::TAG_SPEC ]['extension_spec']['name'] ] = array_pop( $versions );
+		if ( ! isset( $script_spec[ AMP_Rule_Spec::TAG_SPEC ]['extension_spec'] ) ) {
```

Comment:
The Python parser should probably be smarter about extracting all of the extension specs up front to then provide them in a single call, like `AMP_Allowed_Tags_Generated::get_extension_specs()`.

Reply:
I think we should at one point rethink the way the spec is parsed. I've created an issue for that: #3730

Reply:
I _love_ that.

Reply:
While we plan out improving how the spec is parsed, I've implemented a `AMP_Allowed_Tags_Generated::get_extension_specs()` helper to improve things in the meantime. See 9956c90e2d27f2183be41471cf42275f0b51585f.